cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/CompgenCommand.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package io.ballerina.cli.cmd;
+
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import io.ballerina.cli.BLauncherCmd;
+import io.ballerina.cli.launcher.LauncherUtils;
+import io.ballerina.cli.launcher.util.BCompileUtil;
+import io.ballerina.projects.util.ProjectConstants;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static io.ballerina.cli.cmd.Constants.COMPGEN_COMMAND;
+
+/**
+ * This class represents the "bal completion bash-dist" command.
+ *
+ * @since 2.0.0
+ */
+@CommandLine.Command(name = COMPGEN_COMMAND, description = "Generate bash completion script")
+public class CompgenCommand implements BLauncherCmd {
+    private Path userDir;
+    private PrintStream outStream;
+
+    public CompgenCommand() {
+        userDir = Paths.get(System.getProperty(ProjectConstants.USER_DIR));
+        outStream = System.out;
+    }
+
+    public CompgenCommand(Path userDir, PrintStream outStream) {
+        this.userDir = userDir;
+        this.outStream = outStream;
+    }
+
+    @CommandLine.Parameters(description = "Command name")
+    private List<String> bashDistCommands;
+
+    @Override
+    public void execute() {
+        List<String> subCommands;
+        List<String> dirContents = getDirContents(this.userDir);
```

Comment:
Can we load the dir content only if we need them loading it at the top will be a performance hit